src/api/app/components/reports_notice_component.rb

```diff
@@ -0,0 +1,23 @@
+class ReportsNoticeComponent < ApplicationComponent
+  attr_reader :reportable, :user
+
+  def initialize(reportable:, user:)
+    super
+
+    @reportable = reportable
+    @user = user
+  end
+
+  def by_user
+    !!Report.find_by(user:, reportable:)
```

Comment:
```suggestion
    Report.exists?(user:, reportable:)
```
I haven't tested this, but it should work.